src/Catalyst.Core.Modules.Cryptography.BulletProofs.Tests/UnitTests/CryptoWrapperTests.cs

```diff
@@ -0,0 +1,259 @@
+{
+    public sealed class CryptoWrapperTests
+    {
+        public CryptoWrapperTests() { _wrapper = new FfiWrapper(); }
+
+        private readonly ICryptoContext _wrapper;
+        private static readonly Random Random = new Random();
+
+        [Fact]
+        public void TestGenerateKey()
+        {
+            IPrivateKey privateKey = _wrapper.GeneratePrivateKey();
+            privateKey.Bytes.Length.Should().Be(_wrapper.PrivateKeyLength);
+        }
+
+        [Fact]
+        public void TestGenerateDifferentKey()
+        {
+            IPrivateKey privateKey1 = _wrapper.GeneratePrivateKey();
+            IPrivateKey privateKey2 = _wrapper.GeneratePrivateKey();
+            privateKey1.Bytes.Should().NotEqual(privateKey2.Bytes);
+        }
+
+        [Fact]
+        public void TestGetPublicKeyFromPrivate()
+        {
+            IPrivateKey privateKey = _wrapper.GeneratePrivateKey();
+            IPublicKey publicKey = _wrapper.GetPublicKeyFromPrivateKey(privateKey);
+            publicKey.Bytes.Length.Should().Be(_wrapper.PublicKeyLength);
+        }
+
+        [Fact]
+        public void TestStdSignVerify()
+        {
+            IPrivateKey privateKey = _wrapper.GeneratePrivateKey();
+            byte[] message = Encoding.UTF8.GetBytes("fa la la la");
+            byte[] context = Encoding.UTF8.GetBytes("context");
+            ISignature signature = _wrapper.Sign(privateKey, message, context);
+            bool isVerified = _wrapper.Verify(signature, message, context);
+            isVerified.Should().BeTrue();
+        }
+
+        [Fact]
+        public void TestLogicalFailureStdSignVerify()
+        {
+            IPrivateKey privateKey = _wrapper.GeneratePrivateKey();
+            byte[] message1 = Encoding.UTF8.GetBytes("fa la la la");
+            byte[] message2 = Encoding.UTF8.GetBytes("fa la la lahhhhhhh");
+            byte[] context = Encoding.UTF8.GetBytes("any old context");
+            ISignature signature = _wrapper.Sign(privateKey, message1, context);
+            bool isVerified = _wrapper.Verify(signature, message2, context);
+            isVerified.Should().BeFalse();
+        }
+
+        //From https://tools.ietf.org/html/rfc8032#section-7.3
+        [Theory]
+        [InlineData("616263", "98a70222f0b8121aa9d30f813d683f809e462b469c7ff87639499bb94e6dae4131f85042463c2a355a2003d062adf5aaa10b8c61e636062aaad11c2a26083406", "ec172b93ad5e563bf4932c70e1245034c35467ef2efd4d64ebf819683467e2bf", "", true)]
+        [InlineData("616263", "98a70222f0b8121aa9d30f813d683f809e462b469c7ff87639499bb94e6dae4131f85042463c2a355a2003d062adf5aaa10b8c61e636062aaad11c2a26083406", "ec172b93ad5e563bf4932c70e1245034c35467ef2efd4d64ebf819683467e2bf", "a", false)]
+        [InlineData("616261", "98a70222f0b8121aa9d30f813d683f809e462b469c7ff87639499bb94e6dae4131f85042463c2a355a2003d062adf5aaa10b8c61e636062aaad11c2a26083406", "ec172b93ad5e563bf4932c70e1245034c35467ef2efd4d64ebf819683467e2bf", "", false)]
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Split this 241 characters long line (which is greater than 200 authorized).](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4318201)